src/main/java/org/motechproject/mots/scheduler/AutomatedReportJob.java

```diff
@@ -0,0 +1,59 @@
+package org.motechproject.mots.scheduler;
+
+import java.util.Calendar;
+import java.util.Date;
+import org.motechproject.mots.domain.AutomatedReportSettings;
+import org.motechproject.mots.repository.AutomatedReportSettingsRepository;
+import org.quartz.SimpleScheduleBuilder;
+import org.quartz.Trigger;
+import org.quartz.TriggerBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public abstract class AutomatedReportJob extends BaseJob {
+
+  @Autowired
+  private AutomatedReportSettingsRepository automatedReportSettingsRepository;
+
+  public int getIntervalInSeconds(AutomatedReportSettings settings) {
+    if (settings != null && settings.getIntervalInSeconds() != null
```

Comment:
you don't need to check if the setting are null or the IntervalInSeconds is null or > 0